src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1318,11 +1338,245 @@ class Simplex_tree {
     dimension_ = max_dim - dimension_;
   }
 
+  /**
+    * @brief Adds a vertex or an edge in a flag complex, as well as all
+    * simplices of its star, defined to maintain the property
+    * of the complex to be a flag complex, truncated at dimension dim_max.
+    *
+    * @param[in] u,v              Vertex_handle representing the new edge
+    * @param[in] fil              Filtration value of the edge
+    * @param[in] dim_max          Maximal dimension of the expansion.
+    *                             If set to -1, the expansion goes as far as possible.
+    * @param[out] added_simplices Contains at the end all new
+    *                             simplices induced by the insertion of the edge.
+    *                             The container is not emptied and new simplices are
+    *                             appended at the end.
+    *
+    * @pre `SimplexTreeOptions::link_nodes_by_label` must be true.
+    * @pre When inserting the edge `[u,v]`, the vertices @p u and @p v have to be
+    * already inserted in the simplex tree.
+    *
+    * @warning If the edges and vertices are not inserted in the order of their
+    * filtration values, the method `make_filtration_non_decreasing()` has to be
+    * called at the end of the insertions to restore the intended filtration.
+    * @warning Different from other insertion methods, if a vertex or an edges
+    * was already inserted in the simplex, it will be ignored and the filtration
+    * values will not change.
+    */
```

Comment:
A comment about the  call to `clear_filtration()` or to `initialize_filtration()` is required here. Can be with the warning about `make_filtration_non_decreasing`

Reply:
I am not sure what you mean? Like for `filtration_simplex_range`: "If the complex has changed since the last time the filtration was initialized, please call `clear_filtration()` or `initialize_filtration()` to recompute it."?